frontend/src/metabase/hooks/use-synced-query-string.ts

```diff
@@ -27,21 +27,41 @@ export function useSyncedQueryString(
         window.location.pathname + searchString + window.location.hash,
       );
     }
+
+    return () => {
+      // remove the elements of object from the query string when the component is unmounted
+      const search = buildSearchString({
+        filterFn: key => !(key in object),
+      });
+      history.replaceState(
+        null,
+        document.title,
+        window.location.pathname + search + window.location.hash,
+      );
+    };
```

Comment:
Why do we need this now?

Reply:
the test in `e2e/test/scenarios/dashboard-filters/reproductions/16663-filters-can-stay-in-url.cy.spec.js` was failing. tl;dr for the test is that when navigating between dashboards using the command palette, the parameter state wasn't cleared. I'm still trying to figure out exactly what the root cause is, but my hunch is that the `!dashboard` check causes the `SyncedParametersList` to unmount, but because it's using `window.location.search`, it keeps the search state in the URL and doesn't sync with the react router state. This just ensures that if the parameters list does unmount (i.e. if there's no visible parameters or a dashboard doesn't exist) that we "unsync" the parameters if that makes sense

Reply:
I added a code comment to explain and flag this as a workaround for the time being. We can remove the code comments once we've figured out the root cause, or a nicer way of unsyncing parameter lists.